sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,567 @@
+        [(0, 4), (4, 0), (4, 3), (5, 0)]
+
+        """
+        for bcs in m_bcs:
+            self._boundary_conditions['moment'].append(bcs)
+        return self._boundary_conditions['moment']
+
+    def apply_slope_boundary_conditions(self, *s_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_slope_boundary_conditions((4, 3), (5, 0))
+        [(0, 1), (4, 3), (5, 0)]
+
+        """
+        for bcs in s_bcs:
+            self._boundary_conditions['slope'].append(bcs)
+        return self._boundary_conditions['slope']
+
+    def apply_deflection_boundary_conditions(self, *d_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_deflection_boundary_conditions((4, 3), (5, 0))
+        [(0, 2), (4, 3), (5, 0)]
+
+        """
+        for bcs in d_bcs:
+            self._boundary_conditions['deflection'].append(bcs)
+        return self._boundary_conditions['deflection']
+
+
+    def _load_as_SingularityFunction(self, load):
+        x = Symbol('x')
+        if isinstance(load, PointLoad):
+            if load.moment:
+                return load.value*SingularityFunction(x, load.location, S(-2))
+            return load.value*SingularityFunction(x, load.location, S(-1))
+        elif isinstance(load, DistributedLoad):
+            return load.value*SingularityFunction(x, load.start, load.order)
+
+    def apply_loads(self, *loads):
+        """
+        Takes PointLoad and DistributedLoad as input. This method would apply
+        the loads, that are passed as arguments, to the beam object. Internally
+        this method would represent the PointLoads and DistributedLoads using
+        Singularity Functions.
+
+        """
+        for load in loads:
+            if isinstance(load, PointLoad) or isinstance(load, DistributedLoad):
+                self._load += self._load_as_SingularityFunction(load)
+            else:
+                raise TypeError("""apply_loads takes either PointLoad or DistributedLoad objects""")
```

Comment:
It's ok, but, in general, it is bad practice to introduce classes that only have attributes. A dictionary could suffice or namedtuple if you want it a little fancier. I'd rather someone create the class in the future when they actually need attributes and methods, but for now we don't. The api for beam could simply be:

```
my_beam.apply_point_load(x_2, 500)
my_beam.apply_distributed_load(x_4, 100, end=x_5, order=2)
```

Then the user only has to type one line, instead of creating a bunch of load instances. It's just simpler.